Subscription/Subscription.sol

```diff
@@ -240,52 +243,29 @@ contract Subscription {
         returns (bool success)
     {
         // make sure the subscription is valid and ready
-        // pulled this out so I have the hash, should be exact code as "isSubscriptionReady"
-        bytes32 subscriptionHash = getSubscriptionHash(
-            from, to, tokenAddress, tokenAmount, periodSeconds, gasPrice, nonce
-        );
-        address signer = getSubscriptionSigner(subscriptionHash, signature);
+        require(isSubscriptionReady(to, from, tokenAddress, tokenAmount, periodSeconds, gasPrice, nonce, signature), "Subscription is not ready or conditions of transction are not met")
 
-        //make sure they aren't sending to themselves
-        require(to != from, "Can not send to the from address");
-        //the signature must be valid
-        require(signer == from, "Invalid Signature");
-        //timestamp must be equal to or past the next period
+        // now, let make the transfer from the subscriber to the publisher
+        uint256 startingBalance = ERC20(tokenAddress).balanceOf(to);
+        ERC20(tokenAddress).transferFrom(from,to,tokenAmount);
```

Comment:
This function is moved above ln262 to remove the Reentrancy pattern present. Even though the reentrancy is not exploitable it is good practice to not have these patterns present in our contracts.